parser-typechecker/src/Unison/Codebase/TranscriptParser.hs

```diff
@@ -244,20 +242,45 @@ run dir configFile stanzas codebase = do
         output rendered
         when (Output.isNumberedFailure o) $
           if errOk then writeIORef hasErrors True
-          else die
+          else dieWithMsg
         pure numberedArgs
 
-      die = do
+      -- Looks at the current stanza and decides if it is contained in the
+      -- output so far. Appends it if not.
+      appendFailingStanza :: IO ()
+      appendFailingStanza = do
+        stanzaOpt <- readIORef mStanza
+        currentOut <- readIORef out
+        let stnz = maybe "" show (fmap fst stanzaOpt :: Maybe Stanza)
+        unless (stnz `isSubsequenceOf` concat currentOut) $
+          modifyIORef' out (\acc -> acc <> pure stnz)
```

Comment:
This function is required to allow us to overcome the fact that some blocks are hidden, after the fact. When either of `dieWithMsg` or `dieUnexpectedSuccess` gets called, the `awaitInput` method has broken down a stanza, and is working on individual `ucm` commands in the stanza. 

The output so far is read, and if the stanza being worked on and containing an error is already printed (by an `outputEcho` earlier) it is not appended a second time.